rskj-core/src/main/java/co/rsk/cli/tools/DbMigrate.java

```diff
@@ -0,0 +1,211 @@
+            DbKind sourceDbKind,
+            DbKind targetDbKind,
+            String dbName
+    ) {
+        logger.info("Preparing data sources for db: {}", dbName);
+
+        KeyValueDataSource sourceDataSource = KeyValueDataSource.makeDataSource(Paths.get(sourceDbDir, dbName), sourceDbKind);
+        KeyValueDataSource targetDataSource = KeyValueDataSource.makeDataSource(Paths.get(targetDbDir, dbName), targetDbKind);
+
+        logger.info("Data sources prepared successfully");
+        logger.info("Preparing indexes for db: {}", dbName);
+
+        String sourceIndexPath = getIndexDbPath(sourceDbDir, dbName);
+        String targetIndexPath = getIndexDbPath(targetDbDir, dbName);
+
+        logger.info("Indexes prepared successfully");
+        logger.info("Preparing target and source db information...");
+
+        DbInformation sourceDbInformation = new DbInformation(
+                sourceDataSource,
+                sourceIndexPath
+        );
+        DbInformation targetDbInformation = new DbInformation(
+                targetDataSource,
+                targetIndexPath
+        );
+
+        return new DbMigrationInformation(targetDbInformation, sourceDbInformation);
+    }
+
+    private String getIndexDbPath(String databaseDir, String dbName) {
+        return databaseDir + "/" + dbName + "/index";
+    }
+
+    private void migrate(DbMigrationInformation dbMigrationInformation) {
+        KeyValueDataSource sourceKeyValueDataSource = dbMigrationInformation.getSourceDbInformation().getKeyValueDataSource();
+        KeyValueDataSource targetKeyValueDataSource = dbMigrationInformation.getTargetDbInformation().getKeyValueDataSource();
+
+        logger.info("Migrating data source: {}", sourceKeyValueDataSource.getName());
+
+        sourceKeyValueDataSource.keys().forEach(sourceKey ->
+                targetKeyValueDataSource.put(
+                        sourceKey.getData(),
+                        sourceKeyValueDataSource.get(sourceKey.getData())
+                )
+        );
+
+        sourceKeyValueDataSource.close();
+        targetKeyValueDataSource.close();
+
+        logger.info("{} data source migrated successfully", sourceKeyValueDataSource.getName());
+        logger.info("Migrating index for data source: {}", sourceKeyValueDataSource.getName());
+
+        String sourceIndexPath = dbMigrationInformation.getSourceDbInformation().getIndexPath();
+        String targetIndexPath = dbMigrationInformation.getTargetDbInformation().getIndexPath();
+        File sourceIndexFile = new File(sourceIndexPath);
+
+        if (sourceIndexFile.exists() && sourceIndexFile.isFile()) {
+            try {
+                Files.move(Paths.get(sourceIndexPath), Paths.get(targetIndexPath));
+
+                logger.info("Indexes for {} were migrated successfully", sourceKeyValueDataSource.getName());
+            } catch (IOException e) {
+                logger.error("An error happened while migrating indexes", e);
```

Comment:
do you think we could propagate the exception (as is or wrapped) up call stack? Otherwise, it will continue migration process which later will end up with an inconsistent db.

Reply:
Done